packages/ckeditor5-engine/src/model/differ.js

```diff
@@ -336,23 +336,27 @@ export default class Differ {
 	 * @returns {Boolean}
 	 */
 	hasDataChanges() {
+		if ( this._changesInElement.size > 0 ) {
+			return true;
+		}
+
 		for ( const { newMarkerData, oldMarkerData } of this._changedMarkers.values() ) {
 			if ( newMarkerData.affectsData !== oldMarkerData.affectsData ) {
 				return true;
 			}
 
 			if ( newMarkerData.affectsData ) {
-				// Skip markers, which ranges have not changed.
-				if ( newMarkerData.range && oldMarkerData.range && newMarkerData.range.isEqual( oldMarkerData.range ) ) {
-					return false;
```

Comment:
This was incorrect early `false`, it should have been `continue` or changed as I did.